identd.cc

```diff
@@ -260,7 +260,10 @@ void IdentSock::ReadLine(const CS_STRING& line)
 		auto network = sock->GetNetwork();
 		DEBUG("identd: found IRC socket for " + network->GetUser()->GetUserName() + "/" + network->GetName() + " on " + GetLocalIP() + ":" + CString(port));
 
-		reply += "USERID : UNIX : " + network->GetUser()->GetIdent();
+		CString sIdent = network->GetIdent();
```

Comment:
Looks like IRCNetwork::GetIdent() falls back to User::GetIdent() anyway (https://github.com/znc/znc/blob/df6a5073ff9a68bc34981c999ec9ad1e0c4bac09/src/IRCNetwork.cpp#L1486), so we can simplify this a bit:

```
if (sock && sock->GetNetwork())
{
	DEBUG("identd: found IRC socket for " + (network->GetUser() ? network->GetUser()->GetUserName() : "(unknown)") + "/" + network->GetName() + " on " + GetLocalIP() + ":" + CString(port));
	reply += "USERID : UNIX : " + sock->GetNetwork()->GetIdent();
}
```

Reply:
yeah, looks like it works!